src/generate/api.ts

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { model } from "amf-client-js";
+import _ from "lodash";
+
+import {
+  resolveApiModel,
+  parseRamlFile,
+  getAllDataTypes
+} from "../common/parser";
+
+/**
+ * An API represented as an AMF model. Includes the extracted data types
+ * defined in the spec. Common transformations of the name are cached for
+ * reference in templates and for use in assembling file paths.
+ */
+export class Api {
+  dataTypes: model.domain.CustomDomainProperty[];
+  model: model.document.Document;
+  name: {
+    original: string;
+    kebabCase: string;
+    lowerCamelCase: string;
+    snakeCase: string;
+    upperCamelCase: string;
+  };
+  path: string;
+
+  /**
+   * Create a new Api object from a file. This is static and not a constructor
+   * because it is async.
+   *
+   * @param apiSpecFilePath - the path to an API spec file like RAML to be parsed to AMF
+   */
+  static async read(apiSpecFilePath: string): Promise<Api> {
```

Comment:
This seems like a factory pattern and the name should probably reflect that

Reply:
It's factory-ish, but it only ever returns one type of object, so it's not actually. I'm open to different names, but I think it might be confusing to call it factory since it's not really.

Reply:
load / loadFromFile(s)?

Reply:
I like "create" or "make" somewhere in the name, or something that signifies the creation of something is taking place.  Like `createApiFromFile` or just something like `makeApi` or `modelApi` or 'createApiModel`

Reply:
I think that using the class name in the function name is redundant (and uncommon; see `Handlebars.create()`). Using a creation verb is a good idea.